content/blog/readr-2-0-0/index.Rmd

```diff
@@ -0,0 +1,265 @@
+---
+output: hugodown::hugo_document
+
+slug: readr-2-0-0
+title: readr 2.0.0
+date: 2021-07-06
+author: Jim Hester
+description: |
+  This major release of readr includes a new multi-threaded parsing engine powered by vroom and a 
+  number of user interface improvements.
+photo:
+  url: https://unsplash.com/photos/XOW1WqrWNKg
+  author: Anastasia Zhenina
+categories:
+  - package
+---
+
+We're thrilled to announce the release of [readr](https://readr.tidyverse.org/) 2.0.0!
+
+The readr package makes it easy to get rectangular data out of comma separated
+(csv), tab separated (tsv) or fixed width files (fwf) and into R. It
+is designed to flexibly parse many types of data found in the wild, while still
+cleanly failing when data unexpectedly changes.
+
+The easiest way to install the latest version from CRAN is to install the whole tidyverse.
+
+```{r eval = FALSE}
+install.packages("tidyverse")
+```
+
+Alternatively, install just readr from CRAN:
+
+```{r eval = FALSE}
+install.packages("readr")
+```
+
+This blog post will show off the most important changes to the package.
+
+You can see a full list of changes in the [readr release notes](https://github.com/r-lib/readr/releases) and [vroom release notes](https://github.com/r-lib/vroom/releases).
+
+```{r setup}
+library(readr)
+```
+
+## readr edition two
+
+Readr 2.0.0 is a major release of readr and introduces a new second edition parsing and writing engine implemented via the [vroom](https://vroom.r-lib.org/) package.
+
+This engine takes advantage of lazy reading, multi-threading and performance characteristics of modern SSD drives to significantly improve the performance of reading and writing compared to the first edition engine.
+
+We will continue to support the first edition for a number of releases, but eventually this support will be first deprecated and then removed.
+
+You can use the `with_edition()` or `local_edition()` functions to temporarily change the edition of readr for a section of code.
+
+e.g.
+
+- `with_edition(1, read_csv("my_file.csv"))` will read `my_file.csv` with the first edition of readr.
+
+- `readr::local_edition(1)` placed at the top of your function or script will use the first edition for the rest of the function or script.
```

Comment:
```suggestion
readr 2.0.0 is a major release of readr and introduces a new second edition parsing and writing engine implemented via the [vroom](https://vroom.r-lib.org/) package. This engine takes advantage of lazy reading, multi-threading and performance characteristics of modern SSD drives to significantly improve the performance of reading and writing compared to the first edition engine. 

We have done our best to ensure that the two editions parse csv files as similarly as possible, but in case there are differences that affect your code, you can use the `with_edition()` or `local_edition()` functions to temporarily change the edition of readr for a section of code:

- `with_edition(1, read_csv("my_file.csv"))` will read `my_file.csv` with the first edition of readr.

- `readr::local_edition(1)` placed at the top of your function or script will use the first edition for the rest of the function or script.

We will continue to support the 1st edition for a number of releases, but our goal is to ensure that the 2nd edition is uniformly better than the 1st edition so we plan to eventually deprecate and then remove the 1st edition code. 
```